ceph.spec.in

```diff
@@ -888,6 +906,7 @@ cmake .. \
     -DWITH_EMBEDDED=OFF \
     -DWITH_MANPAGE=ON \
     -DWITH_PYTHON3=ON \
+    -DWITH_MGR_DASHBOARD_V2_FRONTEND=OFF \
```

Comment:
why OFF, though?

Reply:
The frontend build requires internet access which is not available in packaging build services, such as OBS. Because of this reason we build the frontend when generating the tarball and include the result files in the tarball.

Reply:
I see. So the frontend files (everything under `src/pybind/mgr/dashboard_v2/frontend/dist`?) are in the tarball, but I don't see them in any RPM or DEB packages?

Reply:
@rjfd yeah, i have the same question. since `make-dist` is updated to include the generated files under `src/pybind/mgr/dashboard_v2/frontend/dist` , we can safely enable this option in `debian/rules` and `ceph.spec.in`, i think?

Reply:
the `WITH_MGR_DASHBOARD_V2_FRONTEND` enables/disables the frontend compilation, if we enable it during the packaging build (inside OBS, or other packaging service) it will require nodejs and npm as system dependencies, and will try to download all nodejs packages frontend dependencies when running `npm install` (https://github.com/ceph/ceph/pull/20103/files#diff-bb5b00ad365f9278caa7bfa071fc7ce3R15). This will fail for two reasons:

a) usually the nodejs and npm system dependencies are too old, that's why we have to upgrade node and npm in install-deps.sh
b) packaging services don't allow internet access during build and `npm install` will fail

This is the reason why we include the frontend compiled files in the tarball to avoid the above problems.

Reply:
It's counter-intuitive for the dashboard to be set to "OFF" in the packaging. There is an analogous situation with Boost, which is downloaded by make-dist. During the build cmake checks for the existence of a Boost-specific file, and only attempts the download if that file doesn't exist. See https://github.com/ceph/ceph/blob/master/cmake/modules/BuildBoost.cmake#L132-L159 for the implementation.

If doing that is too much trouble, could we at least rename the option to something more descriptive (and non-negative) like "-DWITH_MGR_DASHBOARD_V2_NODEJS_DEPS_IN_TARBALL=ON"

Reply:
> If doing that is too much trouble, could we at least rename the option to something more descriptive (and non-negative) like "-
> `DWITH_MGR_DASHBOARD_V2_NODEJS_DEPS_IN_TARBALL=ON"`

From a POV of cmake, this flag disables or enables the build of the frontend. Changing this cmake flag has no effect on the tarball creation.

Reply:
@smithfarm I agree with your concern, and I think we can even drop the cmake option, but that will require another round of build testing that takes some time. Can I open a PR after this one is merged dropping the cmake option?

Reply:
@rjfd Yes, I was going to say that this concern of mine should not block merge of this PR.